enhancements/installer/kubevirt-ipi.md

```diff
@@ -0,0 +1,338 @@
+
+   With that the survey continues to the general cluster name, domain name, and \
+the rest of the non-KubeVirt specific question.
+
+2. Resource creation - terraform
+
+   Terraform uses kubernetes provider to create:
+
+    - DataVolume with RHCOS image \
+         *Note* In disconnected environment the user will need to provide a local image that the installer
+          can upload to the namespace.
+    - secrets for the ignition configs of the VMs
+    - 1 bootstrap machine
+    - 3 masters
+    
+    When installing on pods network:
+    - Services and routes for DNS and LB
+
+3. Bootstrap
+
+    The bootstrap VM has a huge Ignition config set using terraform as secrets and is visible\
+as secrets on the infra OpenShift. KubeVirt boots that VM with that content as ConfigDrive and the 
+bootstraping begins when the `bootkube.service` systemd service starts.\
+
+    This process described more thoroughly in the [installer overview document][https://github.com/OpenShift/installer/blob/37b99d8c9a3878bac7e8a94b6b0113fad6ffb77a/docs/user/overview.md#cluster-installation-process]
+
+4. Masters bootstrap
+
+    Master VMs are booting using a stub Ignition config that are waiting early in\
+the Ignition service to load their Ignition config from a URL. That url is the\
+ `https://<internal-api-vip>/config/master` which is still not available until\
+the **bootstrap** VM is exposing it. It takes few minutes till it does.
+
+    With the machine config available the masters pull their Ignition and boots up\
+joining the tenant cluster as masters and start scheduling pods.
+
+5. Workers bootstrap
+
+    After the masters and the control plane is up, we will scale the machineset to create workers
+    by the machine-api-operator.
+
+
+### Risks and Mitigations
+
+- Network
+
+    - Pods network option
+        -(OCP gap) The ports 22623/22624 that are used by MCS are blocked on the 
+        pods network and prevent from the nodes to pull ignition and updates.
+        - (KubeVirt gap) Interface binding - Currently the only supported binding on the pods
+        network is masquerade which means that all nodes are behind NAT, each VM
+        behind the NAT of his own pod.
+        - (KubeVirt gap) Static IP - Currently the VM egress IP is always the pod IP which is
+        changing every time the VM restarts (and new pod is being created).
+        
+    - Secondary network option (MULTUS)
+    
+        - With this approach admin of the infra cluster will need to be involved in
+    the creation of each new tenant cluster since NADs need to be created and
+    probably also nmstate will need to be used to create the topology on the hosts.
+    At the moment, we will assume that admin created all network resources before running the installer, 
+    and the created networkName (NAD) will be the input for the installer.  
+
+        - Guest-agent is not available at the moment for RHCOS and when Multus is used \
```

Comment:
The reader may not know what "guest-agent" is.

When you say "at the moment" it appears as if you're having a plan to change that. Is that true? Do you have a PR?

It is not clear to me why a guest-agent is needed with the KubeVirt provider but not with other IaaSs. Can you explain?

Reply:
the qemu-guest-agent is needed because without that when running VMs with Multus the hypervisor isn't aware of the IP that the guest allocates from external DHCP.
In KubeVirt, that means that the attribute IP on the VMI resource will be empty.
It's true not just for KubeVirt but for every qemu based virtualization (oVirt/OpenStack).
Today (It's in progress) all these platforms won't show the IP if the guest OS is RHCOS

Reply:
it's not a gap anymore, it was released since I sent this PR

Reply:
> it's not a gap anymore, it was released since I sent this PR

which PR? Can you share a URL?

Reply:
I meant to the current enhancement PR, since I wrote it guest-agent container was released in ERRATA and available now